src/main/resources/db/migration/V1.0.26__update_performance_index_function.sql

```diff
@@ -0,0 +1,161 @@
+union all
+-- Case role permissions
+select
+    j.jurisdiction as jurisdiction,
+    r.region as region,
+    l.location as location,
+    -- Case roles have the case ID specified (no wildcard)
+    l_case_id as case_id,
+    p.role_name as role_name,
+    p.permission as "permission",
+    p.authorization as "authorization",
+    c.classification as classification
+from
+    jurisdictions j,
+    regions r,
+    locations l,
+    case_ids i,
+    -- Use the permissions data for case roles (no authorisations).
+    case_role_permissions p,
+    classifications c) sig;
+return l_signatures;
+end;
+$$;
+
+
+/*
+ * Calculates the signatures of all combined filter conditions which
+ * would match the task.  Each signature is a unique text string which
+ * encodes the filter.  Wildcards are reprsented with '*'.
+ */
+create or replace function cft_task_db.filter_signatures(l_task_id text, l_state cft_task_db.task_state_enum,
+l_jurisdiction text, l_role_category text, l_work_type text, l_region text, l_location text)
+  returns text[] language plpgsql immutable
+as $$
+declare
+l_signatures text[];
+begin
+
+-- The signatures of client filters which will match this task
+select array_agg(
+                   sig.state
+                   || ':' || sig.jurisdiction
+                   || ':' || sig.role_category
+                   || ':' || sig.work_type
+                   || ':' || sig.region
+                   || ':' || sig.location)
+into l_signatures
+from (
+         with
+             states (state) as (select unnest(cft_task_db.add_wildcard(cft_task_db.abbreviate_state(l_state::text)))),
+             jurisdictions (jurisdiction) as (select unnest(cft_task_db.add_wildcard(l_jurisdiction))),
+             role_categories (role_category) as(select unnest(cft_task_db.add_wildcard(cft_task_db.abbreviate_role_category(l_role_category)))),
+             work_types (work_type) as (select unnest(cft_task_db.add_wildcard(l_work_type))),
+             regions (region) as (select unnest(cft_task_db.add_wildcard(l_region))),
+             locations (location) as (select unnest(cft_task_db.add_wildcard(l_location)))
+         select
+             s.state as state,
+             j.jurisdiction as jurisdiction,
+             c.role_category as role_category,
+             w.work_type as work_type,
+             r.region as region,
+             l.location as location
+         from
+             states s,
+             jurisdictions j,
+             role_categories c,
+             work_types w,
+             regions r,
+             locations l) sig;
+return l_signatures;
+end;
+$$;
+
+/*
+ * GIN index to look up tasks based on role signatures and filter signatures.
+ * Only covers assigned and unassigned tasks which have been flagged as
+ * indexed.  There is no value in indexing old tasks, since we never search
+ * them.
+ */
+drop index if exists cft_task_db.search_index;
+create index search_index on cft_task_db.tasks using gin (
```

Comment:
Any way we could reliably reproduce the problem using a functional test or only possible directly with the DB with fill factor etc.

Reply:
Well, We arealy have a functional test that doing the exact same steps, but that didn't have the issue.